src/main/kotlin/no/nav/familie/ba/sak/behandling/vedtak/vedtaksperiode/Opphørsperiode.kt

```diff
@@ -39,6 +41,25 @@ fun mapTilOpphørsperioder(forrigePersonopplysningGrunnlag: PersonopplysningGrun
                 finnOpphørsperioderMellomUtbetalingsperioder(utbetalingsperioder),
                 finnOpphørsperiodeEtterSisteUtbetalingsperiode(utbetalingsperioder)
         ).flatten()
+    }.sortedBy { it.periodeFom }
+
+    return alleOpphørsperioder.fold(mutableListOf(alleOpphørsperioder.first())) { acc: MutableList<Opphørsperiode>, opphørsperiode: Opphørsperiode ->
+        val forrigeOpphørsperiode = acc.last()
+        when {
+            opphørsperiode.periodeFom.isSameOrBefore(forrigeOpphørsperiode.periodeTom ?: TIDENES_ENDE) -> {
+                acc.removeLast()
+                acc.add(opphørsperiode.copy(periodeFom = forrigeOpphørsperiode.periodeFom))
+            }
+            (opphørsperiode.periodeTom ?: TIDENES_ENDE).isSameOrBefore(forrigeOpphørsperiode.periodeTom ?: TIDENES_ENDE) -> {
+                acc.removeLast()
+                acc.add(opphørsperiode.copy(periodeFom = forrigeOpphørsperiode.periodeFom))
+            }
+            else -> {
+                acc.add(opphørsperiode)
+            }
```

Comment:
kan man 
```suggestion
        if (neste.periodeFom.isSameOrBefore(forrige.periodeTom ?: TIDENES_ENDE)) {
                acc.set(acc.lastIndex, forrige.copy(periodeTom =maxOf(forrige.periodeTom, neste.PeriodeTom))
            }
            else -> {
                acc.add(neste)
            }
```

vet ikke om maxOf funker på datoer sånn, men no lignende kanskje